### PR TITLE
Implementation of the detail UI

### PR DESCRIPTION
**Pull request by Juliana-Dimeglio**
Implementation of the detail UI:
   - Error image and Image not available yet
   - When clicking a poster it takes you to the detail of the movie
   -  Button that takes you back to the movies list 

https://user-images.githubusercontent.com/85245831/124652858-fc920c80-de72-11eb-8023-3957cc51bcab.mp4

### CODE REVIEW

File: lib/src/model/result.dart [hunk 1]
```diff
@@ -21,7 +21,7 @@ class Result {
     _voteAverage = result['vote_average'].toString();
     _title = result['title'];
     _popularity = result['popularity'];
-    _posterPath = result['poster_path'];
+    _posterPath = "https://image.tmdb.org/t/p/original" + result['poster_path'];
```
Comment (luisjavierglb) [inline lib/src/model/result.dart]: lets get this path ina constants file please, so we can reuse and change it in just one place

File: lib/src/ui/movie_detail.dart [hunk 2]
```diff
@@ -0,0 +1,139 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'movie_image.dart';
+import '../utils/numeric_constants.dart';
+import '../utils/string_constants.dart';
+import '../utils/styles.dart';
+
+class MovieDetail extends StatelessWidget {
```
Comment (luisjavierglb) [inline lib/src/ui/movie_detail.dart]: `MovieDetailPage` or `MovieDetailScreen` would be a good or better name right?

File: lib/src/ui/movie_list.dart [hunk 3]
```diff
@@ -1,4 +1,7 @@
 import 'package:flutter/material.dart';
+import 'movie_image.dart';
+import 'movie_detail.dart';
+import '../utils/styles.dart';
 import '../utils/numeric_constants.dart';
```
Comment (luisjavierglb) [inline lib/src/ui/movie_list.dart:5]: MovieListState should be private to this file

**Review by luisjavierglb — CHANGES_REQUESTED** [review]
Small changes needed

File: lib/src/model/result.dart [hunk 4]
```diff
@@ -30,10 +30,12 @@ class Result {
     _backdropPath = result['backdrop_path'];
     _adult = result['adult'];
     _overview = result['overview'];
-    _releaseDate = result['release_date'];
+    _releaseDate = result['release_date'] != null
+        ? result['release_date']
+        : "unknown release date";
```
Comment (FranciscoGaraventa) [inline lib/src/model/result.dart]: This can be added to a String file

File: lib/src/ui/movie_image.dart [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import '../utils/numeric_constants.dart';
+import '../utils/string_constants.dart';
+
+class MovieImage extends StatelessWidget {
+  final String posterPath;
+
+  const MovieImage({Key? key, required this.posterPath}) : super(key: key);
```
Comment (FranciscoGaraventa) [inline lib/src/ui/movie_image.dart]: You missed a trailing comma here

Comment (Juliana-Dimeglio) [reply lib/src/ui/movie_image.dart]: Fixed

File: lib/src/model/result.dart (hunk 4 above)
Comment (Juliana-Dimeglio) [reply lib/src/model/result.dart]: Fixed

File: lib/src/model/result.dart (hunk 1 above)
Comment (Juliana-Dimeglio) [reply lib/src/model/result.dart]: Fixed

File: lib/src/ui/movie_detail.dart (hunk 2 above)
Comment (Juliana-Dimeglio) [reply lib/src/ui/movie_detail.dart]: Yes, thank you. Fixed

File: lib/src/ui/movie_list.dart (hunk 3 above)
Comment (Juliana-Dimeglio) [reply lib/src/ui/movie_list.dart:5]: Fixed

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by luisjavierglb — APPROVED** [review]
LGTM

**Pull request merged**